app/src/main/java/com/duckduckgo/app/browser/urlextraction/UrlExtractingWebView.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.urlextraction
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.webkit.WebSettings
+import android.webkit.WebView
+import com.duckduckgo.app.browser.BuildConfig
+import com.duckduckgo.app.browser.useragent.UserAgentProvider
+
+@SuppressLint("SetJavaScriptEnabled", "ViewConstructor")
+class UrlExtractingWebView(
```

Comment:
The `UrlExtractingWebViewClient` is using the same cookie management that's in the `BrowserWebViewClient` (`flushCookies()`, `processUriForThirdPartyCookies()`), I haven't added anything additional regarding this.

Reply:
Can we validate if all the logic we have in the webview and the client needs to be used? I remember during the kickoff we talked about the webview ios was using and their protections. https://app.asana.com/0/492600419927320/1201358138717597/f check the notes here (for example the user agent, and the cookies)

I'm ok with your decision here. But if we are planning to refactor it in the future, it would be nice to document somewhere what's exactly the requirement vs just for convenience.

Reply:
On iOS, it doesn't load images or CSS. There's no obvious way to prevent CSS from loading on Android but `loadsImagesAutomatically` prevents the loading of images.

I've added two subtasks to https://app.asana.com/0/414730916066338/1201729389040240/f that list the functionality that's been used from the browser `WebView` and `BrowserWebClient`.